src/js/pyodide.js

```diff
@@ -147,10 +147,7 @@ function fixRecursionLimit() {
     recurse();
   } catch (err) {}
 
-  let recursionLimit = depth / 60;
-  if (recursionLimit > 1000) {
-    recursionLimit = 1000;
-  }
+  let recursionLimit = Math.min(depth / 50, 400);
```

Comment:
We should do this in a separate PR for traceability. Related to the discussion in #1629

Reply:
Opened #1638.